From 24adc8ff487e68bfaf3d0aa0bc912e8e0ad99aaa Mon Sep 17 00:00:00 2001
From: Chelsio Cudbg <chelsiocudbg@users.noreply.github.com>
Date: Fri, 10 Jul 2026 06:43:44 -0400
Subject: [PATCH 11/11] cxgb4: reduce tc-mqprio max queues to reduce memory
 footprint

Use adapter->tidinfo.uotids.size instead of CXGB4_MAX_ATIDS to
allocate tc-mqprio queues.

adapter->tidinfo.uotids.size is the maximum number of ETHOFLD tids
supported by the adapter.
---
 drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
index 58faf890ec98..9292a8f31eb8 100644
--- a/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
+++ b/drivers/net/ethernet/chelsio/cxgb4/cxgb4_main.c
@@ -6235,13 +6235,11 @@ int cxgb4_adap_probe(struct adapter *adapter)
 
 	for_each_port(adapter, i) {
 		/* For supporting MQPRIO Offload, need some extra
-		 * queues for each ETHOFLD TIDs. Keep it equal to
-		 * MAX_ATIDs for now. Once we connect to firmware
-		 * later and query the EOTID params, we'll come to
-		 * know the actual # of EOTIDs supported.
+		 * queues for each ETHOFLD TIDs.
 		 */
 		netdev = alloc_etherdev_mq(sizeof(struct port_info),
-					   MAX_ETH_QSETS + CXGB4_MAX_ATIDS);
+					   MAX_ETH_QSETS +
+					   adapter->tidinfo.uotids.size);
 		if (!netdev) {
 			err = -ENOMEM;
 			goto out_free_dev;
-- 
2.39.1

